specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/workbooks_API.json

```diff
@@ -0,0 +1,751 @@
+            "schema": {
+              "$ref": "#/definitions/WorkbookErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookAdd": {
+            "$ref": "./examples/WorkbookAdd.json"
+          }
+        }
+      },
+      "patch": {
+        "description": "Updates a workbook that has already been added.",
+        "operationId": "Workbooks_Update",
+        "parameters": [          
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkbookResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "workbookProperties",
+            "description": "Properties that need to be specified to create a new workbook.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Workbook"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The workbook definition updated.",
+            "schema": {
+                "$ref": "#/definitions/Workbook"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookUpdate": {
+            "$ref": "./examples/WorkbookUpdate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/workbooks/{resourceName}": {  
+      "get": {
+        "description": "Get a single workbook by its resourceName.",
+        "operationId": "Workbooks_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkbookResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A workbook definition.",
+            "schema": {
+                "$ref": "#/definitions/Workbook"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/WorkbookErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkbookGet": {
+            "$ref": "./examples/WorkbookGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/workbooks/links": {  
```

Comment:
This URL will not work. There is a segment (workbookName) missing between workbooks and links

Reply:
The other problem with this URL is that it seems like the customer requested to GET resources of type "foo" (links) but what they got was resources of type "bar" (workbooks). This will result in an alert that ARM will open against your service.

Reply:
we'd talked during the last review about removing the links apis entirely because of things like this, and only letting links be modified via PATCH on the workbook itself, making the links there an array instead of just the single initial value.